File: lc1823/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int findTheWinner(int n, int k) {
        List<Integer> persons = new ArrayList<>(n);
        for (int i = 0; i < n; i++) {
            persons.add(i+1);
        }
        int index = 0;
        k--;

        while (persons.size() != 1) {
            index = (index + k) % persons.size();
            persons.remove(index);
        }

        return persons.get(0);
    }
}

File: lc1823/solution.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        persons = [(i+1) for i in range(n)]
        index = 0
        k -= 1
        while len(persons) != 1:
            index = (index + k) % len(persons)
            persons.pop(index)
        
        return persons[0]